uipyqt.py

```diff
@@ -0,0 +1,288 @@
+        self.sc.draw_hand_if_necessary(0)
+        no_more_cards = self.sc.draw_hand_if_necessary(1)
+        if no_more_cards:
+            self.finish_game()
+
+
+    # uncheck any buttons other than the one clicked
+    def disable_all_but_this(self):
+        for btn in self.my_hand_buttons:
+            if not btn is self.sender():
+                btn.setChecked(False)
+
+        self.enable_actions()
+
+    # enable actions: the sum of checked button cards on the table must be equal to
+    # checked card from hand for claiming cards.
+    # No cards on the table and one hand card can be checked for laying card in hand
+    def enable_actions(self):
+        table = []
+        table_card_set = False
+        for table_btn in self.table_buttons:
+            if table_btn.isChecked():
+                table.append(table_btn.text())
+                table_card_set=True
+
+        sum_of_table=tactics.sum_of_cards(table)
+
+        hand_card_set=False
+        hand_card_value=-1
+        for hand_btn in self.my_hand_buttons:
+            if hand_btn.isChecked():
+                hand_card_set=True
+                hand_card_value=int(hand_btn.text()[:2])
+
+        self.lay_card_button.setEnabled(hand_card_set and not table_card_set)
+
+        self.claim_cards_button.setEnabled(hand_card_set and sum_of_table == hand_card_value)
+
+    # again human move
+    def pressed_OK(self):
+        self.my_move()
+
+    # remove card from hand and add it to the table
+    def pressed_lay_card(self):
+        for button in self.my_hand_buttons:
+            if button.isChecked():
+                laid_card = button.text()
+                self.sc.hands[0].remove(laid_card)
+                self.sc.table.append(laid_card)
+                break
+
+        # now it's computer's move
+        self.opponent_move()
+
+    # remove card from hand, cards from the table - and add them to pile
+    def pressed_claim_cards(self):
+        for button2 in self.my_hand_buttons:
+            if button2.isChecked():
+                card_from_hand=button2.text()
+                self.sc.hands[0].remove(card_from_hand)
+                self.sc.piles[0].append(card_from_hand)
+                break
+
+        for button in self.table_buttons:
+            if button.isChecked():
+                card_from_table=button.text()
+                self.sc.table.remove(card_from_table)
+                self.sc.piles[0].append(card_from_table)
+
+        # if table is empty, calculate scopa
+        if len(self.sc.table) == 0:
+            self.sc.scopa_count[0] += 1
+            qmsg = QMessageBox()
+            qmsg.setWindowTitle("Scopa")
+            qmsg.setText(tactics.this_was_scopa)
+            qmsg.exec_()
+
+        self.last_claimed_hand = 0
+        self.opponent_move()
+
+    # if no cards in game claim remaining cards for the player who made the last claim
+    def finish_game(self):
+        for card in self.sc.table.copy():
```

Comment:
Copy of list is not necessary. It's not used later, so its elements don't have to be removed during iterations. To prevent further errors, it can be set to [] after the loop.